src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsDetailsMapping.java

```diff
@@ -201,19 +202,33 @@ public static List<HearingLocation> getHearingLocations(SscsCaseData caseData,
                 .collect(Collectors.toList());
         }
 
-        String epimsId = referenceDataServiceHolder
-            .getVenueService()
-            .getEpimsIdForVenue(caseData.getProcessingVenue())
-            .orElse(null);
+        if (HearingsChannelMapping.isPaperCase(caseData)) {
+            List<VenueDetails> venueDetailsList = referenceDataServiceHolder
+                .getVenueService()
+                .getActiveRegionalEpimsIdsForRpc(caseData.getRegionalProcessingCenter().getName());
 
-        Map<String,List<String>> multipleHearingLocations = referenceDataServiceHolder.getMultipleHearingLocations();
-
-        return multipleHearingLocations.values().stream()
-            .filter(listValues ->  listValues.contains(epimsId))
-            .findFirst()
-            .orElseGet(() -> Collections.singletonList(epimsId))
-            .stream().map(epims -> HearingLocation.builder().locationId(epims).locationType(COURT).build())
-            .collect(Collectors.toCollection(ArrayList::new));
+            return venueDetailsList.stream()
+                .map(VenueDetails::getEpimsId)
+                .map(id -> HearingLocation.builder()
+                    .locationId(id)
+                    .locationType(COURT)
+                    .build())
+                .collect(Collectors.toList());
+        } else {
```

Comment:
You don't need an else here as you return in the if statement and removing will make the code cleaner